File: kotlin/kotlin-vs-java/java-example/src/main/java/org/example/ex04/Exam04.java
package org.example.ex04;

import java.util.HashMap;
import java.util.Map;

/**
 * Map
 */
public class Exam04 {

    public Exam04() {

        var hashMap = new HashMap<String, Object>();
        hashMap.put("key1", "value");
        hashMap.put("key2", 10);

        var map = Map.of(
                "key1", "value",
                "key2", 10
        );
    }

    public static void main(String[] args) {
        new Exam04();
    }
}

File: kotlin/kotlin-vs-java/kotlin-example/src/main/kotlin/ex04/Exam04.kt
package ex04

/**
 * Map
 */
fun main() {
    // Any == Object

    // 불변
    val map = mapOf<String, Any>(
        Pair("key1", "value"),
        "key2" to 10
    )

    // 가변
    val mutableMap = mutableMapOf<String, Any>(
        "key1" to "value"
    )
    mutableMap.set("key2", 10)
    mutableMap["key3"] = "value"

    println(mutableMap["key1"])

    val hashMap = hashMapOf<String, Any>();

    val triple = Triple(first = "", second = "", third = "")
}